[DOC] Create new doc for Dashboard application

vagimeli opened the pull request:
Signed-off-by: vagimeli <vagimeli@amazon.com>

### Description
Create new site page titled Dashboard; create new content on creating your first dashboard

### Issues Resolved
#2349 
#2294 


### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

vagimeli commented:
@joshuarrrr and @abbyhu2000 Please review this first draft of creating dashboards and provide your feedback by Friday, 2/10, if possible.

carolxob commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+# Creating dashboards
+
+A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application you can:
```
```suggestion
A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application, you can:
```

carolxob reviewed (approved):
Looks good, one minor change suggested.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+# Creating dashboards
+
+A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application, you can:
```
Technically, dashboards are not just limited to displaying visualizations (they can render almost anything in a panel), but this is fine as an approximation.

vagimeli replied:
Revised to reflect Dashboards broader capabilities.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application, you can:
+
+- Get started with pre-built panels and dashboards and customize display properties to create the dashboard you need.
```
It seems like this list should either describe the general capabilities of dashboards (what you can do, which would more closely map to the sections of the tutorial, or describe the utility of dashboards (why it's useful). This first item seems like neither. Maybe something along the lines of

```suggestion
- Create a collection of visualizations that can be filtered and customized
```

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application, you can:
+
+- Get started with pre-built panels and dashboards and customize display properties to create the dashboard you need.
+- Analyze, monitor, and alarm on metrics, logs, and traces across multiple data sources.
```
This seems out of scope for a beginner-targeted tutorial, particularly because we haven't yet launched the anywhere project. I'd recommend removing.

vagimeli replied:
Good point. In thinking more braodly about dashboards capabilitites, I've revised this intro.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards/dashboard-index.png" alt="Single-view dashboard in OpenSearch Dashboards" width="600" height="300">
+
+Before you begin this tutorial, make sure you've installed OpenSearch and OpenSearch Dashboards and that you've connected to Dashboards at [http://localhost:5601](http://localhost:5601). The username and password are `admin`.
```
A bit out of scope, but for the target user, I think it's a better assumption to expect that they'll either be doing it on their existing installation of OpenSearch Dashboards, or else on the playground: https://playground.opensearch.org/app/home.  Although neither is perfect, since sample data may not be available on the former, and the latter doesn't support saving.

vagimeli replied:
@joshuarrrr Please confirm the revised text addresses your feedback.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+- The **navigation panel** (A) on the left contains the OpenSearch Dashboards applications.
+- The **toolbar** (B) contains frequently used commands and shortcuts.
+- The **search** bar (C) allows you to search for documents and other objects and add filters.
+- The **time filter** (D) allows you to customize the time and date.
```
If we're having separate items for search and time bar, I'd also include the filter bar (below C), or else include it as part of C.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+- The **search** bar (C) allows you to search for documents and other objects and add filters.
+- The **time filter** (D) allows you to customize the time and date.
+- The **panel** (E) allows you to add existing visualizations to the dashboard or create new ones for the dashboard.
+
```
nit - extra empty line.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+## Adding sample data
+
+If you haven't already added the OpenSearch Dashboards sample datasets as described in [Quickstart guide for OpenSearch Dashboards](https://opensearch.org/docs/latest/dashboards/quickstart-dashboards/), add them before proceeding to the next steps. **Sample eCommerce orders** is used for this tutorial. 
```
Out of scope - we should create an issue about making re-usable snippets for repeated content like this. As we add more pages, it will be increasingly difficult to keep all these copy/pasted sections up to date and in sync.

vagimeli replied:
Good point. Deleted.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+## Creating dashboards
+
+Now that you've added the sample data, you can create a dashboard using that data. The sample dataset has existing sample visualizations, and you can use those visualizations or create new visualizations for the dashboard. For this tutorial, you'll do both.
```
I think we could also have them look at one of the sample dashboards first, and then say the goal of this tutorial will be to create something like that from scratch.

vagimeli replied:
Good point. Change made.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+1. From the navigation panel, choose **Dashboard**.
+2. From the **Dashboards** window, choose **Create Dashboard**.
+3. Set the time filter to **Last 30 days**.
+4. Choose **Add an existing**.
```
This link isn't easy to spot in the entire page

```suggestion
4. From the panel choose **Add an existing**.
```

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+2. From the **Dashboards** window, choose **Create Dashboard**.
+3. Set the time filter to **Last 30 days**.
+4. Choose **Add an existing**.
+5. From the **Add panels** window, choose **Create new** and **Visualization**. This is a shortcut for creating a dashboard and a visualization in one flow. Alternatively, you can use the **Visualize** application to create visualizations and then add them to the dashboard.
+6. From the **New Visualization** window, choose **[eCommerce] Average Sold Quantity**. The donut chart is added to the dashboard.
```
These steps don't seem to agree. In this flow we're just adding an existing visualization, so we should choose (click) `[eCommerce] Average Sold Quantity` from the list (or search for it and click). Are we just wanting them to notice that the `Create new` in this flyout is an alternative to the `Create new` button they'll actually use in the next section?

vagimeli replied:
Thanks for this input. I was mixing the two tasks - adding an existing viz and creating a new viz. Revised this and the following section.

vagimeli replied:
Revised.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+    ![New visualization vertical bar chart]({{site.url}}{{site.baseurl}}/images/dashboards/new-bar-viz.png)
+
+1. Configure the chart's **Data** settings by verifying **Metrics** is **Y-axis Count**.
+1. From the **Buckets** panel, choose **Add** and select **X-axis** from the **Add Bucket** dropdown list.
+1. From the **Aggregation** dropdown list, select **Date histogram**.
+1. Choose **Update**. You'll see a bar chart visualization like the following:
+
+    ![Configuring a visualization]({{site.url}}{{site.baseurl}}/images/dashboards/configure-bar-viz.png)
+
+1. From the toolbar, select **Save**
```
These steps are a bit of a tangent from actually learning about dashboards. For the purpose of the tutorial, it might be better to just save the default vis, without any configurations, so they see the flow of dashboard->vis->dashboard with out extraneous steps.

vagimeli replied:
Revised.

vagimeli replied:
Revised.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+1. In the **Save visualization** window, add a title for the visualization. For example, the title for the bar chart panel is Order Date.
+1. Select **Save and return**.  
+
+The bar chart visualization is now saved and you are taken back to the dashboard. You'll see three visualizations on the dashboard, as shown in the following image: 
```
The screenshot only shows 2?
```suggestion
The bar chart visualization is now saved and you are taken back to the dashboard. You'll see two visualizations on the dashboard, as shown in the following image: 
```

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+![Dashboard showing visualizations combined in a single view]({{site.url}}{{site.baseurl}}/images/dashboards/new-dashboard.png)
+
+## Adding panels
```
maybe?
```suggestion
## Adding subsequent panels
```

otherwise we can just distinguish that using the `Add` option from the toolbar is the more common way to add panels, because the method in the "Creating dashboards" section is _only_ for the case of a completely empty dashboard.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+![Adding another panel to the dashboard]({{site.url}}{{site.baseurl}}/images/dashboards/new-area-viz.png)
+
+## Editing visualizations
```
Should we save the dashboard at this point? We've gone a decent way in an unsaved state, and now might be the time to explain the difference between edit and view modes.

vagimeli replied:
Agreed. Revised.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+## Editing visualizations
+
+You can edit visualizations, configure panels, and specify labels for a panel. For example, you can change the visualization's colors, title, and other attributes. Visualizations can be edited in the panel itself or in edit mode. Only changes made in edit mode are persistent, that is, they are saved in the visualization itself and will be reflected on any dashboard that uses the visualization. Ed ...
```
I think usage of the term "edit mode" here is confusing, particularly because the dashboard itself has an edit mode and a view mode. What we want to distinguish between is tweaking the panel itself vs actually editing the visualization itself. Panel changes include:

1. Toggling visibility of the legend
2. Changing colors via the legend
3. Toggling visibility of the panel title
4. Changing the panel title

I think it might be easier to re-arrange this section to just cover those topics as part of the tutorial, and then mention that it's separate from editing the visualization.

vagimeli replied:
Revised.

vagimeli replied:
@joshuarrrr Will you review this revised section? Do you think we should have a section herein on editing visualizations? Or do you think that tutorial would be better suited to the Building data visualizations section?

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+- To move, select and hold the panel title and drag to the new location.
+- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.
+- To view in fullscreen mode, select the gear icon to open the panel menu and then then select **More > Maximize panel**.
```
I suggest adding a step to mention Minimize as well, so they can get back to the previous state.

vagimeli replied:
Revised.

vagimeli replied:
Added.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+## Filtering documents
+
+Filtering documents is helpful when you want to review data in an ad hoc manner. You can filter documents by applying filters and/or using Dashboards Query Language (DQL). For information about DQL, see [Using Dashboards Query Language]({{site.url}}{{site.baseurl}}/dashboards/discover/dql/). The dashboard queries the data it needs using the filter and/or DQL and applies the filter across the dash ...
```
I'd drop the first sentence. DQL-based searches and filters feel more complex than the rest of this tutorial.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,163 @@
+
+![Saving the dashboard and filter]({{site.url}}{{site.baseurl}}/images/dashboards/save-dashboard.png)
+
+To filter documents using DQL:
+
+1. From the DQL toolbar, specify the search term. For example, input `geoip.region_name: New York` in the query field.
+2. Select **Update**. A dashboard similar to the following is displayed:
+
+![Filtering using DQL]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-dql.png)
```
I'd recommend cutting and saving for a more advanced tutorial.

vagimeli replied:
Agreed. Removed to include in a separate tutorial.

vagimeli replied:
Agreed

joshuarrrr reviewed (commented):
@vagimeli I'm excited for this new tutorial - it will be super helpful for new issues. I had a number of comments, but I'll summarize here

Major issues
1. The "Creating dashboards" flow mixes together the idea of creating a new vis vs using an existing one. My recommendation is to focus on adding an existing vis.
2. The "Editing visualizations" section should be re-organized, with a focus on "Editing panels" instead.
3. Saving shouldn't be all the way at the end, because it's important for users to understand the difference between dashboard view and edit modes.

Smaller suggestions
1. Looking at/linking to a complete sample dashboard first would provide good context for what the tutorial will teach. To avoid writing out steps, we can just deep link to a playground sample dashboard, like [this](https://playground.opensearch.org/app/dashboards#/view/722b74f0-b882-11e8-a6d9-e546fe2bba5f?_g=(filters%3A!()%2CrefreshInterval%3A(pause%3A!f%2Cvalue%3A900000)%2Ctime%3A(from%3Anow-7d%2Cto%3Anow))).
2. I think it's generally odd that we assume the user has a local developer instance of OpenSearch Dashboards running for this type of tutorial. But it's tricky, because their own OSD may not have sample data (or the permissions to add it), and the playground doesn't provide permission to save dashboards or visualizations.
3. It's useful to understand the flow back and forth with the `visualize` app, but we should minimize instructions that happen in the `visualize` app and not on `dashboard`
5. The capabilities section in the intro should better match the tutorial sections
6. DQL filtering can be cut and explained in a more advanced tutorial

vagimeli commented:
> @vagimeli I'm excited for this new tutorial - it will be super helpful for new issues. I had a number of comments, but I'll summarize here
> 
> Major issues
> 
> 1. The "Creating dashboards" flow mixes together the idea of creating a new vis vs using an existing one. My recommendation is to focus on adding an existing vis.
> 2. The "Editing visualizations" section should be re-organized, with a focus on "Editing panels" instead.
> 3. Saving shouldn't be all the way at the end, because it's important for users to understand the difference between dashboard view and edit modes.
> 
> Smaller suggestions
> 
> 1. Looking at/linking to a complete sample dashboard first would provide good context for what the tutorial will teach. To avoid writing out steps, we can just deep link to a playground sample dashboard, like [this](https://playground.opensearch.org/app/dashboards#/view/722b74f0-b882-11e8-a6d9-e546fe2bba5f?_g=(filters%3A!()%2CrefreshInterval%3A(pause%3A!f%2Cvalue%3A900000)%2Ctime%3A(from%3Anow-7d%2Cto%3Anow))).
> 2. I think it's generally odd that we assume the user has a local developer instance of OpenSearch Dashboards running for this type of tutorial. But it's tricky, because their own OSD may not have sample data (or the permissions to add it), and the playground doesn't provide permission to save dashboards or visualizations.
> 3. It's useful to understand the flow back and forth with the `visualize` app, but we should minimize instructions that happen in the `visualize` app and not on `dashboard`
> 4. The capabilities section in the intro should better match the tutorial sections
> 5. DQL filtering can be cut and explained in a more advanced tutorial

@joshuarrrr Thank you for this detailed input. I've revised the tutorial to incorporate your feedback. 

1. Linked to the playground example, as I think it flows better in text. I'm open to your thoughts on how to best show the reader. 
2. Revised the note in the intro to give the reader information about the limitations in the playground or their existing environment. 
3. Removed references to the Visualize application, as users can see the new documentation under that section to learn more. We can revisit the tie in between the Dashboard and Visualize applications once we have this foundational content published.  
4. Deleted the bullets in the intro to simplify the capabilities to date, particularly around alerting and monitoring. 
5. Agreed. Deleted section.  


Will you do another read through of the tutorial? Thanks!

vagimeli commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,167 @@
+
+# Creating dashboards
+
+A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application you can:
```
Revised

ariamarble commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+
+- _Dashboards_ is the abbreviated proper name for OpenSearch Dashboards. OpenSearch Dashboards is an open-source visualization tool designed to work with OpenSearch.
+- _Dashboard_ is the application in OpenSearch Dashboards that is used to track, analyze, and display data at a glance.
+- _dashboard_ or _dashboards_ refer to common names for a tool used to visually display data.
```
should this be "refers" ?

hdhalter replied:
how about _dashboard_ and _dashboards_ are common names for the tools used...

ariamarble commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+
+Showing a legend can give readers more information, while hiding a legend can give it a cleaner look. If you want to show or hide the panel legend:
+
+1. To show the legend, 
```
how do I show the legend?

vagimeli replied:
Added that instruction. Good catch.

ariamarble commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+
+- To move, select and hold the panel title and drag to the new location.
+- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.
+- To view in fullscreen mode, select the gear icon (edit mode) or vertical ellipsis () in the top right of the panel and select **Maximize panel**. To minimize the fullscreen mode, select the vertical ellipsis and select **Minimize**.
```
should the ellipsis be in the parenthesis?

vagimeli replied:
Added

ariamarble reviewed (approved):
Looks good other than my comments. Please review. Thanks!

vagimeli commented:
@kolchfa-aws Thank you for supporting this doc while I'm OOO.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+
+Learn some useful terminology for working with OpenSearch Dashboards and the **Dashboard** application.
+
+- _Dashboards_ is the abbreviated proper name for OpenSearch Dashboards. OpenSearch Dashboards is an open-source visualization tool designed to work with OpenSearch.
+- _Dashboard_ is the application in OpenSearch Dashboards that is used to track, analyze, and display data at a glance.
+- _dashboard_ or _dashboards_ are common names for a tool used to visually display data.
+- _Panel_ is a term used to refer to a visualization displayed on a dashboard. The terms _panel_ and _visualization_ may be used interchangeably throughout this and other Dashboard-related documentation.
```
Helpful! ❤️

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+1. From the dashboard toolbar, choose **Create new**, then **Visualization**.
+1. From the **New Visualization** window, choose **Vertical Bar** and then select the index pattern **opensearch_dashboards_sample_data_ecommerce**. 
+1. From the toolbar, select **Save**
+1. In the **Save visualization** window, add a title for the visualization. For example, the title for the gauge chart panel is [eCommerce] Orders.
+1.  Select **Save and return**.  
+
+The gauge chart visualization is now saved and you are taken back to the dashboard. You'll see two visualizations on the dashboard, like the following: 
+
+![Dashboard showing visualizations combined in a single view]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-combined.png)
```
These steps and the screenshot refer to a gauge chart, but L62 specifies "Vertical Bar", and should be updated to match.

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+
+1. Select the list icon in the panel's lower left corner.
+
+If you want to change the color of the panel legend:
+
+1. Select the gear icon on the area chart panel.
+2. From the **Options** window, select **Edit visualization**.
+3. From the visualization legend, select a category and then select a color from the flyout. The area chart updates with your change.
+4. Select **Save and return**. 
```
Note that it _is_ possible to change chart colors directly from the legend within a dashboard panel. The only difference is that those changes are only saved for that particular dashboard and panel, and don't affect the saved visualization. These steps save the colors on the visualization itself, which will apply to any dashboard that links to the visualization (unless they have already specified different panel colors).

joshuarrrr commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,126 @@
+3. From the visualization legend, select a category and then select a color from the flyout. The area chart updates with your change.
+4. Select **Save and return**. 
+
+If you want to show or hide panel title or customize it without applying the changes across the dashboard overall:
```
"without applying the changes across the dashboard overall" isn't quite right - changing panel titles only affects the particular panel on the particular dashboard, and won't affect any other panel containing that same visualization (or any other dashboard).

joshuarrrr reviewed (commented):
Looking good!

The only major issue is that the instructions mention a "Vertical Bar" chart type where it should be guage.

Other than that, there were a few subtleties of behavior where I tried to provide some clarifying info which can be used or ignored.

natebower commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,142 @@
+
+1. Select the gear icon on the panel.
+2. From the **Options** window, select **Edit panel title**.
+3. From the **Customize panel**, add a title under **Panel title** or toggle the **Show panel title** to hide the title.
```
```suggestion
3. From the **Customize panel**, enter a title under **Panel title** or toggle the **Show panel title** to hide the title.
```

natebower commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,142 @@
+3. From the **Customize panel**, add a title under **Panel title** or toggle the **Show panel title** to hide the title.
+4. Choose **Save**.
+
+Changing panel titles only affects the particular panel on the particular dashboard, and won't affect any other panel containing that same visualization or any other dashboard.
```
```suggestion
Changing panel titles only affects the particular panel on the particular dashboard and won't affect any other panel containing that same visualization or any other dashboard.
```

natebower commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,142 @@
+
+To organize panels, arrange them side-by-side, or resize them, you can use these options:
+
+- To move, select and hold the panel title or the panel top and drag to the new location.
```
```suggestion
- To move a panel, select and hold the panel title or the panel top and drag to the new location.
```

natebower replied:
Is there better terminology than "panel top"?

natebower commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,142 @@
+To organize panels, arrange them side-by-side, or resize them, you can use these options:
+
+- To move, select and hold the panel title or the panel top and drag to the new location.
+- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.
```
```suggestion
- To resize a panel, choose the resize icon in the panel's lower-right corner and drag to the new dimensions.
```

natebower commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,142 @@
+
+- To move, select and hold the panel title or the panel top and drag to the new location.
+- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.
+- To view in fullscreen mode, select the gear icon (edit mode) or vertical ellipsis (⋮) in the top right of the panel and select **Maximize panel**. To minimize the fullscreen mode, select the gear icon or vertical ellipsis and select **Minimize**.
```
```suggestion
- To view a panel in full screen mode, choose the gear icon (edit mode) or vertical ellipsis (⋮) at the top right of the panel and select **Maximize panel**. To minimize the full screen mode, choose the gear icon or vertical ellipsis and select **Minimize**.
```

natebower commented on _dashboards/dashboard/index.md:
```diff
@@ -0,0 +1,142 @@
+- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.
+- To view in fullscreen mode, select the gear icon (edit mode) or vertical ellipsis (⋮) in the top right of the panel and select **Maximize panel**. To minimize the fullscreen mode, select the gear icon or vertical ellipsis and select **Minimize**.
+
+The following is an example of a customized dashboard created in this tutorial.
```
```suggestion
The following is an example of a customized dashboard created by using this tutorial.
```

natebower reviewed (commented):
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!
